pkg/migrations/migrations_test.go

```diff
@@ -103,3 +103,14 @@ func TestCollectFilesFromDir(t *testing.T) {
 		})
 	}
 }
+
+func TestAllNonDeprecatedOperationsAreCreateable(t *testing.T) {
```

Comment:
I think it would be better to make this a compile time constraint for non-deprecated operations by adding

```go
var _ Create = (*OpAddColumn)(nil)
```

to each of the non-deprecated operations.

I can imagine we might forget to add a new operation to `AllNonDeprecatedOperations`, whereas the compile time check would be alongside the existing checks for the `Operation` interface and be less likely to be forgotten